src/Microsoft.Data.Analysis/PrimitiveColumnContainer.cs

```diff
@@ -190,64 +198,69 @@ public void AppendMany(T? value, long count)
-                DataFrameBuffer<byte> mutableLastNullBitMapBuffer = DataFrameBuffer<byte>.GetMutableBuffer(lastNullBitMapBuffer);
-                NullBitMapBuffers[nullBitMapBufferIndex] = mutableLastNullBitMapBuffer;
-                int nullBitMapAllocatable = (int)(((uint)allocatable) / 8) + 1;
-                mutableLastNullBitMapBuffer.EnsureCapacity(nullBitMapAllocatable);
-                _modifyNullCountWhileIndexing = false;
-                for (long i = Length - count; i < Length; i++)
+                remaining -= allocatable;
+                Length += mutableLastBuffer.Length;
+
+                // PR Question: Does this need to be called if it's value is null/doesn't have value?
+                if (value.HasValue)
                 {
-                    SetValidityBit(i, value.HasValue ? true : false);
+                    mutableLastBuffer.RawSpan.Slice(mutableLastBuffer.Length, allocatable).Fill(value ?? default);
+
+                    _modifyNullCountWhileIndexing = false;
+                    for (long i = Length - remaining; i < Length; i++)
+                    {
+                        SetValidityBit(i, value.HasValue);
+                    }
+                    _modifyNullCountWhileIndexing = true;
                 }
-                _modifyNullCountWhileIndexing = true;
-                count -= allocatable;
+
             }
         }
 
         public void ApplyElementwise(Func<T?, long, T?> func)
         {
             for (int b = 0; b < Buffers.Count; b++)
             {
-                ReadOnlyDataFrameBuffer<T> buffer = Buffers[b];
                 long prevLength = checked(Buffers[0].Length * b);
-                DataFrameBuffer<T> mutableBuffer = DataFrameBuffer<T>.GetMutableBuffer(buffer);
-                Buffers[b] = mutableBuffer;
-                Span<T> span = mutableBuffer.Span;
-                DataFrameBuffer<byte> mutableNullBitMapBuffer = DataFrameBuffer<byte>.GetMutableBuffer(NullBitMapBuffers[b]);
-                NullBitMapBuffers[b] = mutableNullBitMapBuffer;
-                Span<byte> nullBitMapSpan = mutableNullBitMapBuffer.Span;
-                for (int i = 0; i < span.Length; i++)
+
+                Span<T> mutableBuffer = Buffers.GetMutable(b).Span;
+                Span<byte> mutableNullBitMapBuffer = NullBitMapBuffers.GetMutable(b).Span;
+
+                for (int i = 0; i < mutableBuffer.Length; i++)
                 {
                     long curIndex = i + prevLength;
-                    bool isValid = IsValid(nullBitMapSpan, i);
-                    T? value = func(isValid ? span[i] : default(T?), curIndex);
-                    span[i] = value.GetValueOrDefault();
-                    SetValidityBit(nullBitMapSpan, i, value != null);
+                    bool isValid = IsValid(mutableNullBitMapBuffer, i);
+                    T? value = func(isValid ? mutableBuffer[i] : default(T?), curIndex);
```

Comment:
Can you use `null` instead of `default(T?)` here? It seems like it would be clearer that this is always null and not a default value of a struct.

Reply:
Yep